pkg/downloader/manager.go

```diff
@@ -246,6 +246,8 @@ func (m *Manager) downloadAll(deps []*chart.Dependency) error {
 	destPath := filepath.Join(m.ChartPath, "charts")
 	tmpPath := filepath.Join(m.ChartPath, "tmpcharts")
 
+	// remove tmpPath after this function returns
+	defer os.RemoveAll(tmpPath)
```

Comment:
I like cleaning up the tmpPath. But, there are cases where this could inadvertently be destructive.

For example, Helm supports a case where a chart in the _charts_ directory does not have an upstream location. But, if MkdirAll on line 264 fails then the tmp location is deleted and those charts could be lost.

On line 271 there is the beginning of a for loop that looks at the dependencies. It has a chance to return part way through the process. A return there causing a removal of the tmp location could be destructive.